client/graph.go

```diff
@@ -56,5 +56,11 @@ type SolveStatus struct {
 
 type SolveResponse struct {
 	// ExporterResponse is also used for CacheExporter
-	ExporterResponse map[string]string
+	ExporterResponse  map[string]string
+	ExporterResponses []ExporterResponse
```

Comment:
SGTM, I think I suggested this before in https://github.com/moby/buildkit/pull/4134#discussion_r1289910449 (but never got around to it :cry:)

While we're here though, we should also split out the `CacheExporterResponses` as well, since those currently suffer from exactly the same issue.

Reply:
I'll focus on the image exporters for now with the tests. Will see if supporting cache responses is something that can be included or done separately.

Reply:
Yeah, atm, all the exporters and cache export results are merged into this one `ExporterResponse` map.

If all the exporters are made their own item in a list, then the caches should be split out too. We definitely *shouldn't* try and merge all the cache exporters into all the `ExporterResponses` (that's possibly the most confusing behavior).